pkg/R/sparkRClient.R

```diff
@@ -2,7 +2,7 @@
 
 # Creates a SparkR client connection object
 # if one doesn't already exist
-connectBackend <- function(hostname, port, timeout = 6000) {
+connectBackend <- function(hostname, port, timeout = 6) {
```

Comment:
@davies So after some debugging in a Ubuntu VM, I found the reason for the travis test failures. This timeout here is not just timeout to make the connection but is the timeout for any messages on the socket. So if we say call `collect` using `callJMethod` then we wait on the socket till there is a reply. Now for long running tasks this may take longer than 6 seconds. I set it to 6000 previously as an upper bound on the time a call will take to return. Its not very ideal, but without having a polling based system or some callback like systems I am not sure its easy to fix. 

Anyways lets not modify this timeout in this PR, we can revisit this later.